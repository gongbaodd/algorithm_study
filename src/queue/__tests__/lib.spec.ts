import { Queue } from "../lib";

describe("Queue", () => {
  it("should create a queue", () => {
    const queue = new Queue<number>();

    expect(queue).toMatchSnapshot();

    expect(queue.top).toBeNull();
    expect(queue.back).toBeNull();
    expect(queue.isEmpty()).toEqual(true);
    expect(queue.size()).toEqual(0);

    queue.enQueue(233);
    expect(queue.top).toEqual(233);
    expect(queue.back).toEqual(233);
    expect(queue.isEmpty()).toEqual(false);
    expect(queue.size()).toEqual(1);

    queue.deQueue();
    expect(queue.top).toBeNull();
    expect(queue.back).toBeNull();
    expect(queue.isEmpty()).toEqual(true);
    expect(queue.size()).toEqual(0);
  });
});
