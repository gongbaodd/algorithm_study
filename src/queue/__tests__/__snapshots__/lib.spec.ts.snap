// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queue should create a queue 1`] = `
Queue {
  "pEnd": null,
  "pHead": null,
}
`;
